packages/common/src/hooks/useAudioRewards.ts

```diff
@@ -32,7 +32,10 @@ export const useFormattedProgressLabel = ({
       label = messages.readyToClaim
     } else if (pending) {
       label = messages.pendingRewards
-    } else if (challenge?.challenge_type === 'aggregate') {
+    } else if (
+      challenge?.challenge_type === 'aggregate' &&
+      challenge?.max_steps !== null
```

Comment:
fwiw I think the challenge type being aggregate makes `max_steps` required, so we could do some coercion instead if you think it's cleaner

I think the "true" type for challenges would be a discriminated union of the different types but that's out of scope for this

Reply:
Yeah, this is my thinking as well. There is a ticket to separate out these types because it's two different cases. This was to get us across the line with the migration without breaking the experience.